pkg/cli/codearea_test.go

```diff
@@ -265,33 +265,83 @@ var codeAreaHandleTests = []HandleTest{
 	{
 		Name: "abbreviation expansion",
 		Given: NewCodeArea(CodeAreaSpec{
-			Abbreviations: func(f func(abbr, full string)) {
+			Abbreviations: func(f func(abbr, full string) bool) {
 				f("dn", "/dev/null")
 			},
 		}),
-		Events:       []term.Event{term.K('d'), term.K('n')},
-		WantNewState: CodeAreaState{Buffer: CodeBuffer{Content: "/dev/null", Dot: 9}},
+		Events:       []term.Event{term.K('d'), term.K('n'), term.K(' ')},
+		WantNewState: CodeAreaState{Buffer: CodeBuffer{Content: "/dev/null ", Dot: 10}},
 	},
 	{
-		Name: "abbreviation expansion preferring longest",
+		Name: "abbreviation expansion",
```

Comment:
Some early comments on the test cases (I haven't reviewed them in detail since they will change a bit after you act on my other comments):

* All test cases should have unique names describing the facet of the behavior they test against.

* Not sure why you removed the test against the behavior of preferring the longest candidate.